Thu Nov 27 2003  Arnaud Charlet  <charlet@adacore.com>

	* configure: Add missing G_DISABLE_CAST_CHECKS when
	using --disable-debug
	Submitted at http://bugzilla.gnome.org/show_bug.cgi?id=331994
	Committed on Feb 21 2006

	* pango/pango-layout.c: Do not call utf8_validate when assertions
	are disabled.
	Submitted at http://bugzilla.gnome.org/show_bug.cgi?id=331995
	Committed on Feb 21 2006

--- configure	2003/11/26 13:59:18	1.1
+++ configure	2003/11/26 13:59:29
@@ -20247,7 +20247,7 @@
   PANGO_DEBUG_FLAGS="-DG_DISABLE_CAST_CHECKS"
 
   if test "x$enable_debug" = "xno"; then
-    PANGO_DEBUG_FLAGS="$GLIB_DEBUG_FLAGS -DG_DISABLE_ASSERT -DG_DISABLE_CHECKS"
+    PANGO_DEBUG_FLAGS="$GLIB_DEBUG_FLAGS -DG_DISABLE_ASSERT -DG_DISABLE_CHECKS $PANGO_DEBUG_FLAGS"
   fi
 fi
 
--- pango/pango-layout.c	2003/11/26 14:16:48	1.1
+++ pango/pango-layout.c	2003/11/22 14:37:36
@@ -823,7 +823,11 @@ pango_layout_set_text (PangoLayout *layo
 
   old_text = layout->text;
 
+#ifdef G_DISABLE_ASSERT
+  if (length == -1)
+#else
   if (length != 0)
+#endif
     {
       if (!g_utf8_validate (text, length, &end))
 	g_warning ("Invalid UTF8 string passed to pango_layout_set_text()");
